src/Workspaces/Remote/ServiceHub/Services/LanguageServer.cs

```diff
@@ -0,0 +1,169 @@
+    {
+        public CSharpLanguageServer(Stream stream, IServiceProvider serviceProvider)
+            : base(stream, serviceProvider, LanguageNames.CSharp)
+        {
+        }
+    }
+
+    internal class VisualBasicLanguageServer : LanguageServer
+    {
+        public VisualBasicLanguageServer(Stream stream, IServiceProvider serviceProvider)
+            : base(stream, serviceProvider, LanguageNames.VisualBasic)
+        {
+        }
+    }
+
+    internal abstract class LanguageServer : ServiceBase
+    {
+        private static readonly IImmutableSet<string> s_supportedKinds =
+            ImmutableHashSet.Create(
+                NavigateToItemKind.Class,
+                NavigateToItemKind.Constant,
+                NavigateToItemKind.Delegate,
+                NavigateToItemKind.Enum,
+                NavigateToItemKind.EnumItem,
+                NavigateToItemKind.Event,
+                NavigateToItemKind.Field,
+                NavigateToItemKind.Interface,
+                NavigateToItemKind.Method,
+                NavigateToItemKind.Module,
+                NavigateToItemKind.Property,
+                NavigateToItemKind.Structure);
+
+        private readonly string _languageName;
+
+        public LanguageServer(Stream stream, IServiceProvider serviceProvider, string languageName)
+            : base(serviceProvider, stream, SpecializedCollections.EmptyEnumerable<JsonConverter>())
+        {
+            _languageName = languageName;
+
+            StartService();
+        }
+
+        [JsonRpcMethod(Methods.InitializeName)]
+        public object Initialize(int? processId, string rootPath, Uri rootUri, ClientCapabilities capabilities, TraceSetting trace, CancellationToken cancellationToken)
+        {
+            // our LSP server only supports WorkspaceStreamingSymbolProvider capability
+            // for now
+            return new InitializeResult()
+            {
+                Capabilities = new VSServerCapabilities()
+                {
+                    WorkspaceStreamingSymbolProvider = true
+                }
+            };
+        }
+
+        [JsonRpcMethod(Methods.InitializedName)]
+        public Task Initialized()
+        {
+            return Task.CompletedTask;
+        }
+
+        [JsonRpcMethod(Methods.ShutdownName)]
+        public void Shutdown(CancellationToken cancellationToken)
+        {
+            // our language server shutdown when VS shutdown
+            // we have this so that we don't get log file every time VS shutdown
+        }
+
+        [JsonRpcMethod(Methods.ExitName)]
+        public void Exit()
+        {
+            // our language server exit when VS shutdown
+            // we have this so that we don't get log file every time VS shutdown
+        }
+
+        [JsonRpcMethod(VSSymbolMethods.WorkspaceBeginSymbolName)]
+        public Task<VSBeginSymbolParams> WorkspaceBeginSymbolAsync(string query, int searchId, CancellationToken cancellationToken)
```

Comment:
>WorkspaceBeginSymbolAsync [](start = 41, length = 25)

`WorkspaceBeginSymbolAsync` name does not make any sense to me. Shouldn't this be something like `BeginSymbolSearchAsync`?

Reply:
@gundermanc ?

I guess he followed existing LSP pattern. anyway, I can change method name since protocol doesn't care what actual implementation's name is as long as it is marked with JsonRpcMethod and signature is right.